actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -334,7 +334,7 @@ def raw_post
     # variable is already set, wrap it in a StringIO.
     def body
       if raw_post = get_header("RAW_POST_DATA")
-        raw_post = raw_post.dup.force_encoding(Encoding::BINARY)
+        (+raw_post).force_encoding(Encoding::BINARY)
```

Comment:
If `raw_post` is frozen, `(+raw_post)` will return new instance then `raw_post` will lose `force_encoding(Encoding::BINARY)`.

Reply:
Would it be enough to switch to
```ruby
raw_post = (+raw_post).force_encoding(Encoding::BINARY)
```

Reply:
```suggestion
        raw_post = (+raw_post).force_encoding(Encoding::BINARY)
```